src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -3242,8 +3241,11 @@ private List<String> getArtifactsAbsolutePath(JavadocPathArtifact javadocArtifac
 
             DependencyFilter filter = new ScopeDependencyFilter(
                     Arrays.asList(Artifact.SCOPE_COMPILE, Artifact.SCOPE_PROVIDED), Collections.emptySet());
-            DependencyRequest req =
-                    new DependencyRequest(new DefaultDependencyNode(RepositoryUtils.toArtifact(artifact)), filter);
+            DependencyRequest req = new DependencyRequest(
```

Comment:
Can we add an IT for doclets?

Reply:
feel free to take over the PR, I verified it against the user provided reproducer in MJAVADOC-742